packages/peregrine/lib/talons/MyAccount/useNewsletterSubscription.js

```diff
@@ -0,0 +1,73 @@
+import { useCallback, useState } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import { useToasts } from '@magento/peregrine';
+import { useUserContext } from '../../context/user';
+import { useAwaitQuery } from '../../hooks/useAwaitQuery';
+
+export const useNewsletterSubscription = props => {
+    const { subscribeMutation, customerQuery } = props;
+    const [{ currentUser, isSignedIn }, { getUserDetails }] = useUserContext();
+
+    const [subscribe, setSubscribe] = useState(
+        currentUser && currentUser.hasOwnProperty('is_subscribed')
+            ? currentUser.is_subscribed
+            : false
+    );
+    const [isSubmitting, setIsSubmitting] = useState(false);
+
+    const [subscription, { error: subscriptionError }] = useMutation(
+        subscribeMutation,
+        {
+            fetchPolicy: 'no-cache'
+        }
+    );
+    const fetchUserDetails = useAwaitQuery(customerQuery);
+    const [, { addToast }] = useToasts();
+
+    const errors = [];
+    if (subscriptionError) {
+        errors.push(subscriptionError.graphQLErrors[0]);
+    }
+
+    const toggleSubscription = useCallback(() => {
+        setSubscribe(!subscribe);
+    }, [subscribe, setSubscribe]);
+
+    const handleSubmit = useCallback(
+        async ({ is_subscribed }) => {
+            setIsSubmitting(true);
+            try {
```

Comment:
This submission doesn't fire a callback on success so you shouldn't need this try/catch. Errors are automatically logged by an `onError` link, so that's also not needed anymore.